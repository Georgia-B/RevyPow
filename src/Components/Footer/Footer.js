import React from 'react';
import SubscribeSwitch from '../SubscribeSwitch/SubscribeSwitch';

const Footer = (props) => {
    const { hasPermission, data, unsubscribeUser, subscribeUser } = props
    return (
        <div className="footer">
            <SubscribeSwitch isSubscribed={hasPermission} onClick={hasPermission ? unsubscribeUser : subscribeUser} />
            {data && data.dateUpdated &&
                <div className="footer__date">
                    <div>Last updated</div>
                    <div>{data.dateUpdated.value}</div>
                </div>
            }
        </div>
    )
}

export default Footer;