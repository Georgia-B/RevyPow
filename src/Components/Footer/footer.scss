.footer {
    position: fixed;
    bottom: 0;
    left: 0;
    width: 100%;
    height: 10vh;
    display: flex;
    align-items: center;
    flex-direction: row;

    &__date {
        text-align: right;
        flex: 1;
        padding-right: 30px;
    }
}

@media (max-width: $breakpoint-medium) {
    .footer__date {
        display: none;
    }
}