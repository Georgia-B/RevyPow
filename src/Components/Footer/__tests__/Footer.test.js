import React from 'react'
import { cleanup, render } from '@testing-library/react'
import Footer from '../Footer';

describe('Components | <Footer /> when user subscribed', () => {
    afterEach(cleanup);
    it('should render correctly', () => {
        const component = render(<Footer
            hasPermission={true}
            data={{ dateUpdated: "yesterday" }}
            subscribeUser={() => true}
            unsubsribeUser={() => true}
        />);
        expect(component.asFragment()).toMatchSnapshot();
    })
})

describe('Components | <Footer /> when user not subscribed', () => {
    afterEach(cleanup);
    it('should render correctly', () => {
        const component = render(<Footer
            hasPermission={false}
            data={{ dateUpdated: "yesterday" }}
            subscribeUser={() => true}
            unsubsribeUser={() => true}
        />);
        expect(component.asFragment()).toMatchSnapshot();
    })
})