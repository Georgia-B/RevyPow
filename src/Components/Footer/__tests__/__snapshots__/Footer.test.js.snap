// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components | <Footer /> when user not subscribed should render correctly 1`] = `
<DocumentFragment>
  <div
    class="footer"
  >
    <div
      class="subscribe__container"
    >
      <label
        class="MuiFormControlLabel-root"
      >
        <span
          class="MuiSwitch-root"
        >
          <span
            aria-disabled="false"
            class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-93 MuiSwitch-switchBase MuiSwitch-colorSecondary"
          >
            <span
              class="MuiIconButton-label"
            >
              <input
                aria-label="primary checkbox"
                class="PrivateSwitchBase-input-96 MuiSwitch-input"
                type="checkbox"
                value="false"
              />
              <span
                class="MuiSwitch-thumb"
              />
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </span>
          <span
            class="MuiSwitch-track"
          />
        </span>
        <span
          class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
        >
          Notifications
        </span>
      </label>
    </div>
    <div
      class="footer__date"
    >
      <div>
        Last updated
      </div>
      <div />
    </div>
  </div>
</DocumentFragment>
`;

exports[`Components | <Footer /> when user subscribed should render correctly 1`] = `
<DocumentFragment>
  <div
    class="footer"
  >
    <div
      class="subscribe__container"
    >
      <label
        class="MuiFormControlLabel-root"
      >
        <span
          class="MuiSwitch-root"
        >
          <span
            aria-disabled="false"
            class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-19 MuiSwitch-switchBase MuiSwitch-colorSecondary PrivateSwitchBase-checked-20 Mui-checked"
          >
            <span
              class="MuiIconButton-label"
            >
              <input
                aria-label="primary checkbox"
                checked=""
                class="PrivateSwitchBase-input-22 MuiSwitch-input"
                type="checkbox"
                value="true"
              />
              <span
                class="MuiSwitch-thumb"
              />
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </span>
          <span
            class="MuiSwitch-track"
          />
        </span>
        <span
          class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
        >
          Notifications
        </span>
      </label>
      <div
        class="subscribe__message"
      >
        You are subscribed to the morning snow report.
      </div>
    </div>
    <div
      class="footer__date"
    >
      <div>
        Last updated
      </div>
      <div />
    </div>
  </div>
</DocumentFragment>
`;
